pulp_rpm/tests/functional/api/test_checkpoint.py

```diff
@@ -0,0 +1,186 @@
+"""Tests for checkpoint distribution and publications."""
+
+from datetime import datetime, timedelta
+import re
+from time import sleep
+from urllib.parse import urlparse
+from aiohttp import ClientResponseError
+from tempfile import NamedTemporaryFile
+import pytest
+import requests
+from pulp_rpm.tests.functional.constants import RPM_SIGNED_URL
+from pulpcore.content.handler import Handler
+
+
+@pytest.fixture(scope="class")
+def rpm_package_factory_class(
+    gen_object_with_cleanup,
+    pulp_domain_enabled,
+    rpm_package_api,
+):
+    """Return a Package created from uploading an RPM file."""
+
+    def _rpm_package_factory_class(url=RPM_SIGNED_URL, pulp_domain=None):
+        with NamedTemporaryFile() as file_to_upload:
+            file_to_upload.write(requests.get(url).content)
+            file_to_upload.flush()
+            upload_attrs = {"file": file_to_upload.name}
+
+            kwargs = {}
+            if pulp_domain:
+                if not pulp_domain_enabled:
+                    raise RuntimeError("Server does not have domains enabled.")
+                kwargs["pulp_domain"] = pulp_domain
+
+            return gen_object_with_cleanup(rpm_package_api, **upload_attrs, **kwargs)
+
+    return _rpm_package_factory_class
+
+
+@pytest.fixture(scope="class")
+def setup(
+    rpm_repository_factory,
+    rpm_publication_factory,
+    rpm_distribution_factory,
+    rpm_package_factory_class,
+    rpm_repository_api,
+    monitor_task,
+):
+    def create_publication(repo, checkpoint):
+        artifact = rpm_package_factory_class()
+        monitor_task(
+            rpm_repository_api.modify(
+                repo.pulp_href, {"add_content_units": [artifact.pulp_href]}
+            ).task
+        )
+        return rpm_publication_factory(repository=repo.pulp_href, checkpoint=checkpoint)
+
+    repo = rpm_repository_factory()
+    distribution = rpm_distribution_factory(repository=repo.pulp_href, checkpoint=True)
+
+    pubs = []
+    pubs.append(create_publication(repo, False))
+    sleep(1)
```

Comment:
We're already waiting for the tasks to complete in `create_publication()`, why is the manual sleep needed?

Reply:
Could you write this in a comment to make it clear from reading the test?  Otherwise I could imagine someone thinking it's just a race condition hack